### PR TITLE
Reverse history fix

### PR DESCRIPTION
**Pull request by JulianKarlBauer**
Prior to this fix, there had been two problems:
1. The history did not include the "count" of the `cube_move` (Actually, all moves had `count=1`)
2. I think we expect 
   ```
   import magiccube
   cube = magiccube.Cube(3, "YYYYYYYYYRRRRRRRRRGGGGGGGGGOOOOOOOOOBBBBBBBBBWWWWWWWWW")
   print(cube)
   
   cube.rotate("R2 L")
   cube.rotate(cube.reverse_history())

   print(cube)
   ```
   to yield
   ```bash
            Y  Y  Y                   
            Y  Y  Y                   
            Y  Y  Y                   
   R  R  R  G  G  G  O  O  O  B  B  B 
   R  R  R  G  G  G  O  O  O  B  B  B 
   R  R  R  G  G  G  O  O  O  B  B  B 
            W  W  W                   
            W  W  W                   
            W  W  W                   
  
            Y  Y  Y                   
            Y  Y  Y                   
            Y  Y  Y                   
   R  R  R  G  G  G  O  O  O  B  B  B 
   R  R  R  G  G  G  O  O  O  B  B  B 
   R  R  R  G  G  G  O  O  O  B  B  B 
            W  W  W                   
            W  W  W                   
            W  W  W  
   ```
   however, [v0.2.10](https://github.com/trincaog/magiccube/releases/tag/v0.2.10) yields
  
   ```bash
            Y  Y  Y                   
            Y  Y  Y                   
            Y  Y  Y                   
   R  R  R  G  G  G  O  O  O  B  B  B 
   R  R  R  G  G  G  O  O  O  B  B  B 
   R  R  R  G  G  G  O  O  O  B  B  B 
            W  W  W                   
            W  W  W                   
            W  W  W                   
  
            Y  Y  G                   
            Y  Y  G                   
            Y  Y  G                   
   R  R  R  G  G  W  O  O  O  Y  B  B 
   R  R  R  G  G  W  O  O  O  Y  B  B 
   R  R  R  G  G  W  O  O  O  Y  B  B 
            W  W  B                   
            W  W  B                   
            W  W  B  
   ```

### CODE REVIEW

**Comment by JulianKarlBauer** [comment]
@trincaog please excuse the messy Git history, I had to fight with pyenv and my autoformatter-settings. Please feel free to squash the result. Do you think this is a beneficial contribution? Does it induce any problems on solvers not yet covered by the tests?

File: magiccube/cube_move.py [hunk 1]
```diff
@@ -74,12 +74,12 @@ class CubeMove():
     regex_pattern = re.compile(
         "^(?:([0-9]*)(([LRDUBF])([w]?)|([XYZMES]))([']?)([0-9]?))$")
 
-    def __init__(self, move_type: CubeMoveType, is_reversed: bool = False, wide: bool = False, layer: int = 1):
+    def __init__(self, move_type: CubeMoveType, is_reversed: bool = False, wide: bool = False, layer: int = 1, count=1):
```
Comment (trincaog) [inline magiccube/cube_move.py:77]:
```suggestion
    def __init__(self, move_type: CubeMoveType, is_reversed: bool = False, wide: bool = False, layer: int = 1, count: int = 1):
```

**Review by trincaog — COMMENTED** [review]
Thank you so much contributing, this is a great improvement 👏 

Left a minor request to add type hints to the new parameter (let me know if you'd rather have me do it).

Also, I'm curious on the issues you had building the library. If you have suggestions to improve developer experience on the library, it would be great.

**Comment by JulianKarlBauer** [comment]
Thank you for the type hint improvement! If you are satisfied with the changes, please feel free to merge. If you want me to prepare a release due to this minor change, feel free to ask for it.

For now, I think the developer experience is good and does not need any improvements.

The small issues I faced initially are:
I usually work with either conda or venv.
While following your detailed instructions on `pyenv`, I had to install `3.11` before being able to use the command `pyenv virtualenv 3.11 cube-3.11`.
However, on my local machine I then ran into a problem while installing dependencies `pip install -r requirements.txt` which I now can't reproduce. So this is probably due to local package inconsistency...

Eventually I just worked with `venv` and temporarily replaced `version = "{{VERSION_PLACEHOLDER}}"` with `"0.0.1"` to install magiccube as the tests run against the installed package.

**Review by trincaog — APPROVED** [review]
Thank you for the fix and for the tip on the build process.
I'll try to reproduce it to understand it better.

It looks great 💪

**Pull request merged**